### PR TITLE
fireworks[major]: switch to pydantic v2

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/fireworks/langchain_fireworks/llms.py [hunk 1]
```diff
@@ -39,8 +35,21 @@ class Fireworks(LLM):
 
     base_url: str = "https://api.fireworks.ai/inference/v1/completions"
     """Base inference API URL."""
-    fireworks_api_key: SecretStr = Field(default=None, alias="api_key")
-    """Fireworks AI API key. Get it here: https://fireworks.ai"""
+    fireworks_api_key: SecretStr = Field(
```
Comment (baskaryan) [inline libs/partners/fireworks/langchain_fireworks/llms.py:38]: @hwchase17

File: libs/partners/fireworks/langchain_fireworks/chat_models.py [hunk 2]
```diff
@@ -369,32 +371,32 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
...
-            values["client"] = Fireworks(**client_params).chat.completions
-        if not values.get("async_client"):
-            values["async_client"] = AsyncFireworks(**client_params).chat.completions
-        if values["max_retries"]:
-            values["client"]._max_retries = values["max_retries"]
-            values["async_client"]._max_retries = values["max_retries"]
-        return values
+        if not (self.client or None):
+            self.client = Fireworks(**client_params).chat.completions
+        if not (self.async_client or None):
```
Comment (eyurtsev) [inline libs/partners/fireworks/langchain_fireworks/chat_models.py:394]: if not self.async_client:

File: libs/partners/fireworks/langchain_fireworks/chat_models.py [hunk 3]
```diff
@@ -369,32 +371,32 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
...
 
-        if not values.get("client"):
-            values["client"] = Fireworks(**client_params).chat.completions
-        if not values.get("async_client"):
-            values["async_client"] = AsyncFireworks(**client_params).chat.completions
-        if values["max_retries"]:
-            values["client"]._max_retries = values["max_retries"]
-            values["async_client"]._max_retries = values["max_retries"]
-        return values
+        if not (self.client or None):
```
Comment (eyurtsev) [inline libs/partners/fireworks/langchain_fireworks/chat_models.py:392]: if not self.client:

File: libs/partners/fireworks/langchain_fireworks/chat_models.py (hunk 2 above)
Comment (baskaryan) [reply libs/partners/fireworks/langchain_fireworks/chat_models.py:394]:
```suggestion
        if not self.async_client:
```

File: libs/partners/fireworks/langchain_fireworks/chat_models.py (hunk 3 above)
Comment (baskaryan) [reply libs/partners/fireworks/langchain_fireworks/chat_models.py:392]:
```suggestion
        if not self.client:
```

**Pull request merged**